src/app/item-details/item-details.component.ts

```diff
@@ -1,15 +1,26 @@
-import { Component, OnInit } from '@angular/core';
+import { DaysLeftCountedPipe } from './../pipes/daysLeftCountedPipe/days-left-counted.pipe';
+import { ActionItem } from './../services/tasksService/tasks.service';
+import { Component, Input, ChangeDetectionStrategy, OnChanges } from '@angular/core';
 
 @Component({
   selector: 'app-item-details',
   templateUrl: './item-details.component.html',
-  styleUrls: ['./item-details.component.css']
+  styleUrls: ['./item-details.component.css'],
+  providers: [DaysLeftCountedPipe],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class ItemDetailsComponent implements OnInit {
+export class ItemDetailsComponent implements OnChanges {
+  @Input() item: ActionItem;
+  @Input() id: string;
+  @Input() errorMessage: string | undefined;
+  @Input() isLoadingActionItem: boolean;
+  @Input() currentDate: Date;
+  itemVisibility: boolean;
 
-  constructor() { }
-
-  ngOnInit() {
+  ngOnChanges() {
+    if (this.item) {
+      const itemDueDate = this.item.dueDate;
+      this.itemVisibility = typeof itemDueDate === 'string' ? false : true;
```

Comment:
If this property is responsible for keeping information about the visibility of due date then the name of this property is wrong.

Also, you don't need to use `typeof` operator:
`this.itemVisibility = !!itemDueDate`
should be enough.

Reply:
This one is still not addressed.